latest/SSC/CGI/keywords/cgi_states.yaml

```
@@ -20,9 +20,115 @@ properties:
     title: FSM profile
     description: The FSM profile
     type: string
+    maxLength: 20
+    enum:
+      - NFOV
+      - EFOV
+      - SPEC660
+      - SPEC730
+      - EXCAM
+      - SPARE
+      - FSM_PROFILE_UNKNOWN
+    archive_catalog:
+      datatype: nvarchar(20)
+      destination: [CGIExposure.fsmprofile, CGIMosaic.fsmprofile]
+  OPGAIN:
+    title: The commanded gain (can be designated "Auto" to be calculated after HOWFSC)
+    description:
+      The commanded gain (can be designated "Auto" to be calculated after
+      HOWFSC)
+    anyOf:
+      - type: string
+      - type: float
     maxLength: 32
```

```suggestion
      - type: string
         maxLength: 32
      - type: number
```
`maxLength` needs to be directly with the `type: string` because it only applies to it.
There is no `type: float` in json-schema it is `type: number` instead

---

Thanks- fixed in latest commit